pkg/socialevent/socialevent_name_generator.go

```diff
@@ -0,0 +1,30 @@
+package socialevent
+
+import (
+	"crypto/rand"
+	"math/big"
+	"strings"
+)
+
+// NewName SocialEvent names (a.k.a, activation codes) are generated in a standard format,
+// 4 blocks of 4 case insensitive, alphanumeric characters separated by a minus sign delimiter, like so:
+// XXXX-XXXX-XXXX-XXXX
```

Comment:
This looks like an overkill :)
Let's make it a bit more user friendly.
For example use digest only and limit the number of digests to 6-8. Extra security is good but we can address it differently/later.

Reply:
addressed in f8390ff